src/schemas/ThreeIdKeychain.json

```diff
@@ -2,30 +2,37 @@
   "$schema": "http://json-schema.org/draft-07/schema#",
   "definitions": {
     "JWE": {
+      "title": "JWE",
       "type": "object",
       "properties": {
-        "protected": {
-          "type": "string"
-        },
-        "iv": {
-          "type": "string"
-        },
-        "ciphertext": {
-          "type": "string"
-        },
-        "tag": {
-          "type": "string"
-        },
+        "protected": { "type": "string" },
+        "iv": { "type": "string" },
+        "ciphertext": { "type": "string" },
+        "tag": { "type": "string" },
+        "aad": { "type": "string" },
         "recipients": {
-          "type": "array"
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "header": {
+                "type": "object",
+                "properties": {
+                  "alg": { "type": "string" },
+                  "iv": { "type": "string" },
+                  "tag": { "type": "string" },
+                  "epk": { "type": "object" },
+                  "kid": { "type": "string" }
+                },
+                "required": ["alg", "iv", "tag"]
```

Comment:
Please don't include this, all JWEs don't necessarily have these properties. Just allow any object for the header.

Reply:
Aren't `alg`, `iv` and `tag` required? wouldn't it be a risk of runtime errors trying to decrypt invalid headers?

Reply:
They are not required for JWE, but they are required for the Xchacha20poly1305 with X25519 key exchange. We can't enforce them at this layer though because that restricts which algos we can use here in the future.